File: samples/dotnet-isolated-azurefunction/InputBinding/StateInputBinding.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

namespace dotnet_isolated_azurefunction
{
    using Microsoft.Azure.Functions.Worker;
    using Microsoft.Azure.Functions.Worker.Extensions.Dapr;
    using Microsoft.Azure.Functions.Worker.Http;
    using Microsoft.Extensions.Logging;

    public static class StateInputBinding
    {
        [Function("StateInputBinding")]
        public static string Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "state/{key}")] HttpRequestData req,
            [DaprStateInput("statestore", Key = "{key}")] string state, 
            FunctionContext functionContext)
        {
            var log = functionContext.GetLogger("StateInputBinding");
            log.LogInformation("C# HTTP trigger function processed a request.");

            return state;
        }
    }
}

File: samples/javascript-v4-azurefunction/src/functions/stateInputBinding.js
const { app, input, trigger } = require('@azure/functions');

const daprStateInput = input.generic({
    type: "daprState",
    stateStore: "%StateStoreName%",
    direction: "in",
    name: "order",
    key: "{key}"
});

app.generic('StateInputBinding', {
    trigger: trigger.generic({
        type: 'httpTrigger',
        authLevel: 'anonymous',
        methods: ['GET'],
        route: "state/{key}",
        name: "req"
    }),
    extraInputs: [daprStateInput],
    return: output.generic({
        type: "http",
    }),
    handler: async (request, context) => {
        context.log("Node HTTP trigger function processed a request.");

        const daprStateInputValue = context.extraInputs.get(daprStateInput);
        // print the fetched state value
        context.log(daprStateInputValue);

        return { body: daprStateInputValue };
    }
});